### PR TITLE
homepage rework and other improvements

### PR DESCRIPTION
**Pull request by juhp**
- rework the homepage layout and content
- enable navbar for homepage
- more navbar entries
- lts-16
- drop /?page
- hoogle-5.0.18
- improve /snapshots "buttons"
- improve snapshot top matter
- [q] better to use haskellstack.org for "get started"?

Opening this now for early review, comments, and feedback?
(I think it may need a little more hamlet/yesod foo, but I am reasonably happy so far.)

### CODE REVIEW

**Comment by juhp** [comment]
Screenshot from 2020-10-29 13-49-23

**Comment by juhp** [comment]
- (Btw I noticed for stackage.org, when I go above 200% zoom, the navbar entries go away)
- (Also stackage.org text rendering is pretty hideous on Chrome (I am using chromium) in dark mode)

**Comment by juhp** [comment]
With normal default light mode desktop (120%)

Screenshot from 2020-10-29 14-04-53

File: src/Handler/Home.hs [hunk 1]
```diff
@@ -30,12 +30,8 @@ getHomeR :: Handler Html
 getHomeR = track "Handler.Snapshots.getAllSnapshotsR" $ do
     cacheSeconds $ 60 * 60
     now' <- getCurrentTime
-    currentPageMay <- lookupGetParam "page"
```
Comment (snoyberg) [inline src/Handler/Home.hs:33]: IIUC, this isn't a good idea. We're already having trouble with long database queries, isn't this disabling paging?

File: templates/default-layout.hamlet [hunk 2]
```diff
@@ -1,17 +1,15 @@
 <div class="navbar navbar-inverse navbar-static-top">
   <div class="navbar-inner">
     <div class="container">
-      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
-        <span class="icon-bar">
-        <span class="icon-bar">
-        <span class="icon-bar">
       <a class="brand" href="/">
-        <img src="/static/img/stackage.png" title="FP Complete">
```
Comment (snoyberg) [inline templates/default-layout.hamlet:9]: Hah, good catch!

File: templates/default-layout.hamlet [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
+        <img src="/static/img/stackage.png" title="Stackage">
       <div class="nav-collapse collapse">
         <ul class="nav">
-          $forall route <- [AllSnapshotsR, BlogHomeR]
-            <li :Just route == cur:.active>
-              <a href=@{route}>#{route}
+            <li> <a href="/#about">about</a>
```
Comment (snoyberg) [inline templates/default-layout.hamlet:8]:
```suggestion
            <li> <a href="/#about">About</a>
```

**Review by snoyberg — COMMENTED** [review]
Looking good! I wonder if the homepage is slightly too wordy with too much small-ish text. What do you think about collapsing to one column, increasing visibility of the important stuff, and leaving the more verbose "about" content to the end? Just an idea

**Comment by juhp** [comment]
> Looking good! I wonder if the homepage is slightly too wordy with too much small-ish text.

I do agree - I was hesitating to delete too much text in one (or first) go. :-)
(So I tried "hiding" it at the bottom as a first attempt.)

> What do you think about collapsing to one column,

You mean not putting LTS and Snapshots side by side?

> increasing visibility of the important stuff, and leaving the more verbose "about" content to the end?

Yeah, we can try that.

File: src/Handler/Home.hs (hunk 1 above)
Comment (juhp) [reply src/Handler/Home.hs:33]: I may be wrong, this is just for dropping the redundant https://www.stackage.org/?page=2 "feature".

**Comment by juhp** [comment]
How about this now?:

Updated screenshot 133%
Screenshot from 2020-10-30 21-37-20

Still has 2 columns... but I add a summary at the top
(also removed links from the 2 section titles)
More details in the last commit.

**Comment by juhp** [comment]
Seems, I broke the mobile navbar menu earlier...

**Comment by juhp** [comment]
An alternative to Recent News could also be to just put the Blog post headline below the logo.

Another simplifying idea could be merging Recent News and Recent Snapshots (just listing the latest lts and nightly snapshot).

**Comment by juhp** [comment]
Okay, @snoyberg, rows are gone now, and I combined few Recent Snapshots into News.

Apart from the mobile nav menu, I feel this is close to ready now.

With some placeholder snapshot texts to make it clearer:
Screenshot from 2020-10-31 20-37-55

I rather like this now.

**Comment by juhp** [comment]
Okay I restored the navbar mobile menu too.

(I feel it kicks in too wide (early), but I think that is in the (modified) bootstrap library?)

File: src/Handler/Home.hs (hunk 1 above)
Comment (snoyberg) [reply src/Handler/Home.hs:33]: Doh, my mistake, sorry. Thanks for clarifying.

**Comment by snoyberg** [comment]
I think this is an improvement. My recommendation: I merge/release this, and then we iterate with others on further improvements as desired. What do you think?

Thanks for attacking this, I really like the changes!

**Comment by juhp** [comment]
@snoyberg sounds good to me: I am all for further incremental improvements.

Might be good just to test it for a real build: though hopefully it works as expected.
